play-zipkin-tracing/play26/src/main/scala/jp/co/bizreach/trace/play26/filter/ZipkinTraceFilter.scala

```diff
@@ -27,31 +32,64 @@ import scala.util.Failure
 class ZipkinTraceFilter @Inject() (tracer: ZipkinTraceServiceLike)(implicit val mat: Materializer) extends Filter {
 
   import tracer.executionContext
-  private val reqHeaderToSpanName: RequestHeader => String = ZipkinTraceFilter.ParamAwareRequestNamer
+
+  // TODO: any way to get rid of this asInstanceOf?
+  private val handler: HttpServerHandler[RequestHeader, Result] =
+    HttpServerHandler.create(tracer.httpTracing, new HttpAdapter)
+      .asInstanceOf[HttpServerHandler[RequestHeader, Result]]
+  private val extractor: TraceContext.Extractor[Headers] =
+    tracer.httpTracing.tracing.propagation.extractor(ZipkinTraceFilter.headerGetter)
 
   def apply(nextFilter: (RequestHeader) => Future[Result])(req: RequestHeader): Future[Result] = {
-    val serverSpan = tracer.serverReceived(
-      spanName = reqHeaderToSpanName(req),
-      span = tracer.newSpan(req.headers)((headers, key) => headers.get(key))
-    )
-    val result = nextFilter(req.withHeaders(new Headers(
-      (req.headers.toMap.mapValues(_.headOption getOrElse "") ++ tracer.toMap(serverSpan)).toSeq
-    )))
+    var template = req.attrs.get(Router.Attrs.HandlerDef).map(_.path)
+    // TODO ^^ is always None! is it being read too soon?
```

Comment:
`HandlerDef` is supplied by Play from the parsed routes file. If we use a custom router, it isn't unavailable.

Reply:
@simao last place waiting for answer was here. Do you know how we can handle custom router? We should ideally have tests to cover both routes file and custom router